Engine/RealTime/BacktestingRealTimeHandler.cs

```diff
@@ -123,13 +136,19 @@ public void ScanPastEvents(DateTime time)
         {
             foreach (var scheduledEvent in GetScheduledEventsSortedByTime())
             {
+                if (scheduledEvent.NextEventUtcTime > time)
+                {
+                    // since they're sorted, we don't need to go through every single one
+                    break;
```

Comment:
Isn't this already covered by the `while` below? The `_sortingScheduledEventsRequired = true;` might go in the while loop

Reply:
This statement tells us to break from the `foreach` loop, so if you have 1M scheduled events and they're _all_ in the future, we only check the first one. This is safe because they're already chronologically sorted. 

Regarding the sorted flag, I do believe you're correct. If the current event's time equals the provided `time`, then we won't fire any events but we would end up sorting again. Great eye.